Facebook.iOS/CoreKit/source/Facebook.CoreKit/ApiDefinition.cs

```diff
@@ -128,184 +128,437 @@ interface AccessToken : Copying, INSSecureCoding {
+		[Wrap ("AppEventParameterName.MaxRatingValue")]
+		NSString MaxRatingValueParameterNameKey { get; }
+
+		[Obsolete ("Use the AppEventParameterName.NumItems property instead. This will be removed in future versions.")]
+		[Wrap ("AppEventParameterName.NumItems")]
+		NSString NumItemsParameterNameKey { get; }
+
+		[Obsolete ("Use the AppEventParameterName.PaymentInfoAvailable property instead. This will be removed in future versions.")]
+		[Wrap ("AppEventParameterName.PaymentInfoAvailable")]
+		NSString PaymentInfoAvailableParameterNameKey { get; }
+
+		[Obsolete ("Use the AppEventParameterName.RegistrationMethod property instead. This will be removed in future versions.")]
+		[Wrap ("AppEventParameterName.RegistrationMethod")]
+		NSString RegistrationMethodParameterNameKey { get; }
+
+		[Obsolete ("Use the AppEventParameterName.SearchString property instead. This will be removed in future versions.")]
+		[Wrap ("AppEventParameterName.SearchString")]
+		NSString SearchStringParameterNameKey { get; }
+
+		[Obsolete ("Use the AppEventParameterName.Success property instead. This will be removed in future versions.")]
+		[Wrap ("AppEventParameterName.Success")]
+		NSString SuccessParameterNameKey { get; }
+
+		[Obsolete ("Use the AppEventParameterValue.Yes property instead. This will be removed in future versions.")]
+		[Wrap ("AppEventParameterValue.Yes")]
+		NSString YesParameterValueKey { get; }
+
+		[Obsolete ("Use the the AppEventParameterValue.No property instead. This will be removed in future versions.")]
+		[Wrap ("AppEventParameterValue.No")]
 		NSString NoParameterValueKey { get; }
 
-		// extern NSString *const FBSDKAppEventParameterNameAdType;
-		[Field ("FBSDKAppEventParameterNameAdType", "__Internal")]
+		[Obsolete ("Use the AppEventParameterName.AdType property instead. This will be removed in future versions.")]
+		[Wrap ("AppEventParameterName.AdType")]
 		NSString AdTypeParameterNameKey { get; }
 
-		// extern NSString *const FBSDKAppEventParameterNameOrderID;
-		[Field ("FBSDKAppEventParameterNameOrderID", "__Internal")]
+		[Obsolete ("Use the AppEventParameterName.OrderId property instead. This will be removed in future versions.")]
+		[Wrap ("AppEventParameterName.OrderId")]
 		NSString OrderIdParameterNameKey { get; }
 
 		// +(void)logEvent:(NSString *)eventName;
 		[Static]
 		[Export ("logEvent:")]
+		void LogEvent (NSString eventName);
+
+		[Static]
+		[Wrap ("var ptr = NSString.CreateNative (eventName);\nLogEvent (NSString.FromHandle (ptr));\nNSString.ReleaseNative (ptr)")]
 		void LogEvent (string eventName);
 
 		// +(void)logEvent:(NSString *)eventName valueToSum:(double)valueToSum;
 		[Static]
 		[Export ("logEvent:valueToSum:")]
+		void LogEvent (NSString eventName, double valueToSum);
+
+		[Static]
+		[Wrap ("var ptr = NSString.CreateNative (eventName);\nLogEvent (NSString.FromHandle (ptr), valueToSum);\nNSString.ReleaseNative (ptr)")]
```

Comment:
No, this is not ok, you should just do `new NSString (eventName)` you are basically:

1. creating a boat to send your cargo from Cambridge side to Boston side of the Charles river
2. sending the boat alone to Boston side and you are walking over the bridge to the Boston side carrying your cargo
3. destroying your not needed boat on the Boston side.

you just did not need the boat at all (`NSString.CreateNative` and `NSString.ReleaseNative`) you just need to walk the bridge by doing `new NSString (eventName)`.